src/main/java/echoserver/EchoServerService.java

```diff
@@ -0,0 +1,33 @@
+package echoserver;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.util.ArrayList;
+
+public class EchoServerService {
+
+
+    public EchoServerService(HostServer hostServer, Logger logger) {
+        this.hostServer = hostServer;
+        this.logger = logger;
+    }
+
+    public void start() throws IOException {
+        ClientConnection connectedClient;
+
+        while ((connectedClient = hostServer.listenForClientConnection()) != null) {
+            Thread newConnection = new Thread(new EchoServerServiceThread(connectedClient, logger));
+            clients.add(connectedClient);
+            newConnection.start();
+        }
+    }
+
+    public void stop() throws IOException {
+        clients.forEach((client) -> client.close());
+        hostServer.close();
```

Comment:
This service be responsible for closing the `hostServer` and the `clientConnections`?

Reply:
It looks like I don't need to do garbage collection for clients. I looked back at the error that caused me to manually close the clients, and the error was from the tests, not the actual code. So, I removed it.